src/Atc.Kepware.Configuration/Services/Connectivity/KepwareConfigurationClientConnectivity.cs

```diff
@@ -472,6 +502,95 @@ public sealed partial class KepwareConfigurationClient
             cancellationToken);
     }
 
+    private static HttpClientRequestResult<IList<TDevice>?> ProcessGetDevicesByChannelNameResponse<TDevice>(HttpClientRequestResult<IList<JsonObject>?> response)
+        where TDevice : DeviceBase
+    {
+        // No data, return early, nothing to adapt
+        if (!response.HasData)
+        {
+            return new HttpClientRequestResult<IList<TDevice>?>()
+            {
+                CommunicationSucceeded = response.CommunicationSucceeded,
+                StatusCode = response.StatusCode,
+                Message = response.Message,
+                Exception = response.Exception,
+            };
+        }
+
+        // No actual devices, return early, nothing to adapt
+        if (response.Data!.Count == 0)
+        {
+            return new HttpClientRequestResult<IList<TDevice>?>([])
+            {
+                CommunicationSucceeded = response.CommunicationSucceeded,
+                StatusCode = response.StatusCode,
+                Message = response.Message,
+                Exception = response.Exception,
+            };
+        }
+
+        if (!TryGetDeviceTypeJsonMappingType<TDevice>(out Type? jsonMappingType))
+        {
+            return new HttpClientRequestResult<IList<TDevice>?>()
+            {
+                CommunicationSucceeded = response.CommunicationSucceeded,
+                StatusCode = response.StatusCode,
+                Message = response.Message,
+                Exception = new NotSupportedException($"Could not find a JSON mapping type for {typeof(TDevice).Name}"),
+            };
+        }
+
+        // Serialize to JSON mapping type
```

Comment:
Havent tested it, but couldnt we simplify this code to the following ?

```csharp
        var deserializedObjects = response
            .Data
            .Select(x => JsonSerializer.Deserialize(x.ToString(), jsonMappingType, jsonSerializerOptions))
            .ToList();

        var typedDeviceList = deserializedObjects as IList<TDevice>;
```

Reply:
Nope, `ToList()` returns a value of type `List<object?>` and can not be cast to `IList<TDevice>`.

Reply:
And `jsonMappingType` will be `...KepwareContracts.Connectivity.<Driver>.<TDevice>`, not `...Contracts.Connectivity.Drivers.<Driver>.<TDevice>`

Reply:
The follwoing does work though:

``` csharp
IList<TDevice> deviceTypes = response
    .Data
    .Select(x => JsonSerializer.Deserialize(x.ToString(), jsonMappingType))
    .Select(x => x.Adapt<TDevice>())
    .ToList();
```

Reply:
Guess we still would want the added JsonSerializerOptions added:

.Select(x => JsonSerializer.Deserialize(x.ToString(), jsonMappingType, jsonSerializerOptions))